src/nodes.ts

```diff
@@ -20,10 +20,11 @@ import { GraphDatabaseNode } from './graph/graphNodes';
 
 type Experience = "MongoDB" | "DocumentDB" | "Graph" | "Table";
 
-export interface IDocumentNode extends INode {
+export interface IDocument {
```

Comment:
Minor stylistic change subject to modification - 
Decouple the `IDocumentNode` from `INode`. This makes it more clear that `DocumentEditor` does not need a `vscode.TreeItem` structure to operate on. Moreover, a previous implementation with the `IDocumentNode` extending `INode` left unused properties in `MongoDummyNode` just to satisfy interface implementation.

Reply:
This is unnecessary if you remove MongoDummyNode